### PR TITLE
Feature/views with hooks

### PR DESCRIPTION
**Pull request by Gustav-Eikaas**
# Description

Please include a summary of the change and which issue is fixed.
List any external dependencies that are required for this change.

Completes: AB#FILL_IN_YOUR_ISSUE_ID

## Type of change

Remove options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code.
- [ ] I have linked my DevOps task using the AB# tag.
- [ ] My code is easy to read.
- [ ] Documentation and comments is added where needed.
- [ ] My code is covered by tests.
- [ ] UX has reviewed relevant UI contributions.

### CODE REVIEW

File: client/packages/portal-client/src/components/portal-header/Header.tsx [hunk 1]
```diff
@@ -1,29 +1,25 @@
...
   useMenuItems,
-  useNavigateBasedOnWorkSurface,
-  useWorkSurface,
 } from '@equinor/portal-core';
 import { useNavigate } from 'react-router-dom';
 
 export function Header() {
-  const currentWorkSurface = useCurrentWorkSurface();
-  const { setCurrentWorkSurface } = useWorkSurface();
+  const { getId, setViewId } = useCurrentWorkSurfaceId();
```
Comment (Gustav-Eikaas) [inline client/packages/portal-client/src/components/portal-header/Header.tsx:10]: rename view

File: client/packages/portal-client/src/components/portal-header/Header.tsx [hunk 2]
```diff
@@ -1,29 +1,25 @@
...
-  //Preload all menu items
-  useNavigateBasedOnWorkSurface();
   useMenuItems();
   return (
     <PortalHeader
       onLogoClick={() => {
-        if (location.pathname.includes('/apps') && currentWorkSurface) {
-          navigate(`/${currentWorkSurface.name}`);
+        const id = getId();
+        if (location.href.includes('/apps') && id) {
```
Comment (Gustav-Eikaas) [inline client/packages/portal-client/src/components/portal-header/Header.tsx:17]: appIsMounted

File: client/packages/portal-client/src/components/portal-header/Header.tsx [hunk 3]
```diff
@@ -1,29 +1,26 @@
...
   const navigate = useNavigate();
-  //Preload all menu items
-  useNavigateBasedOnWorkSurface();
   useMenuItems();
   return (
     <PortalHeader
       onLogoClick={() => {
-        if (location.pathname.includes('/apps') && currentWorkSurface) {
-          navigate(`/${currentWorkSurface.name}`);
+        const id = getId();
```
Comment (Noggling) [inline client/packages/portal-client/src/components/portal-header/Header.tsx:17]: Kan du lage en named funksjon her f.eks handleOnLogoClick?

File: client/packages/portal-core/src/avatar/TopBarAvatar.tsx [hunk 4]
```diff
@@ -26,21 +22,19 @@ export const TopBarAvatar = (): JSX.Element | null => {
...
-  if (!user) return null;
+  if (isLoading) {
+    return (
+      <Icon
+        color={tokens.colors.interactive.primary__resting.hex}
+        name="account_circle"
+      />
+    );
+  }
+  if (error || !data) return <div>Error</div>;
```
Comment (Noggling) [inline client/packages/portal-core/src/avatar/TopBarAvatar.tsx:35]: Her må vi er på og få startet på en standard komponent for error

File: client/packages/portal-core/src/currentWorkSurfaceContext/CurrentViewContext.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+import { LoadingWorkSurfacesTransition } from './LoadingWorkSurfacesTransition';
+
+/** View context type */
+type ViewController = {
+  setViewId: (viewId: string | undefined) => void;
+  getId: () => string | undefined;
+  id$: Observable<string | undefined>;
+};
+
+const CurrentWorkSurfaceId = createContext<ViewController>({} as any);
```
Comment (Noggling) [inline client/packages/portal-core/src/currentWorkSurfaceContext/CurrentViewContext.tsx:24]: CurrentViewId

File: client/packages/portal-core/src/currentWorkSurfaceContext/CurrentViewContext.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+export const useCurrentWorkSurfaceId = () => useContext(CurrentWorkSurfaceId);
```
Comment (Noggling) [inline client/packages/portal-core/src/currentWorkSurfaceContext/CurrentViewContext.tsx:26]: useCurrentViewId

File: client/packages/portal-core/src/types/portal-config.ts [hunk 7]
```diff
@@ -33,7 +33,7 @@ export interface PortalConfig {
   agGrid?: {
     licenseKey: string;
   };
-  phases?: WorkSurface[];
+  phases?: View[];
```
Comment (Noggling) [inline client/packages/portal-core/src/types/portal-config.ts:36]: views

File: client/packages/portal-pages/src/pages/phase-page/PhasePage.tsx [hunk 8]
```diff
@@ -1,30 +1,23 @@
...
-  const surface =
-    module.getCurrentWorkSurface() ??
-    module?.getWorkSurfaces()?.find((s) => s.name === workSurfaceKey);
 
-  if (!surface) return <div>Phase not found</div>;
+  const { data, isLoading } = useViews();
+  if (isLoading) return <div>Loading....</div>;
+  if (!data) return <div>Something went wrong</div>;
+  const phase = data.find((s) => s.name === workSurfaceKey);
+  if (!phase) return <div>Phase not found</div>;
```
Comment (Noggling) [inline client/packages/portal-pages/src/pages/phase-page/PhasePage.tsx:14]: View not found

Comment (Noggling) [reply client/packages/portal-pages/src/pages/phase-page/PhasePage.tsx:14]: bare rename filen også så og component

File: client/packages/portal-pages/src/pages/phase-page/PhasePageHeader.tsx [hunk 9]
```diff
@@ -32,7 +32,7 @@ export const StyledIconWrapper = styled.div<{ active?: boolean }>`
     0px 5px 22px rgba(0, 0, 0, 0.12), 0px 7px 8px rgba(0, 0, 0, 0.2);
 `;
 
-export const PasePageHeader = ({ name, shortName, subtext, icon }: Phase) => {
+export const PasePageHeader = ({ name, shortName, subtext, icon }: View) => {
```
Comment (Noggling) [inline client/packages/portal-pages/src/pages/phase-page/PhasePageHeader.tsx:35]: ViewPageHeader

File: client/packages/portal-pages/tsconfig.lib.json [hunk 10]
```diff
@@ -19,5 +19,11 @@
     "**/*.spec.jsx",
     "**/*.test.jsx"
   ],
-  "include": ["**/*.js", "**/*.jsx", "**/*.ts", "**/*.tsx"]
+  "include": [
+    "**/*.js",
```
Comment (Noggling) [inline client/packages/portal-pages/tsconfig.lib.json:23]: bør vi fjerne js og jsx?

File: client/packages/portal-utils/src/app/appMounted.ts [hunk 11]
```diff
@@ -0,0 +1,7 @@
+export function shouldSuspense() {
+  return !appMounted();
+}
+
+export function appMounted() {
```
Comment (Noggling) [inline client/packages/portal-utils/src/app/appMounted.ts:5]: 👍

File: client/packages/portal-core/src/queries/portal/getMenuItems.ts [hunk 12]
```diff
@@ -0,0 +1,13 @@
+import { IHttpClient } from '@equinor/fusion-framework-module-http';
+import { View } from '../../types/view';
+
+/** Get menu items based on current view id */
+export async function getMenuItems(
```
Comment (Noggling) [inline client/packages/portal-core/src/queries/portal/getMenuItems.ts:5]: getViewById

File: client/packages/portal-pages/src/pages/home-page/HomePage.tsx [hunk 13]
```diff
@@ -7,26 +9,23 @@ import {
...
 
 export const HomePage = (): JSX.Element => {
-  const navigate = useNavigate();
-
-  const module = useWorkSurface();
-
+  const { isLoading, data } = useViews();
+  const { setViewId } = useViewController();
+  if (isLoading) return <FullPageLoading detail="Loading views" />;
+  if (!data) return <div>Something went wrong</div>;
```
Comment (Noggling) [inline client/packages/portal-pages/src/pages/home-page/HomePage.tsx:17]: Lag to todo!!

Comment (Noggling) [reply client/packages/portal-pages/src/pages/home-page/HomePage.tsx:17]: ok kort i devops

File: client/packages/portal-pages/src/pages/phase-page/PhasePage.tsx (hunk 8 above)
Comment (Noggling) [reply client/packages/portal-pages/src/pages/phase-page/PhasePage.tsx:15]: lag todo og kort i devops

File: client/packages/portal-pages/src/pages/phase-page/PhasePage.tsx [hunk 14]
```diff
@@ -1,30 +1,24 @@
...
-    return <></>;
-  }
-  const surface =
-    module.getCurrentWorkSurface() ??
-    module?.getWorkSurfaces()?.find((s) => s.name === workSurfaceKey);
-
-  if (!surface) return <div>Phase not found</div>;
+  const { data, isLoading } = useViews();
+  if (isLoading) return <FullPageLoading detail="Loading phase" />;
+  if (!data) return <div>Something went wrong</div>;
```
Comment (Noggling) [inline client/packages/portal-pages/src/pages/phase-page/PhasePage.tsx:13]: todo og kort i devops

File: client/packages/portal-pages/src/pages/phase-page/PhasePage.tsx [hunk 15]
```diff
@@ -1,30 +1,24 @@
...
+  const phase = data.find((s) => s.name === viewKey);
+  if (!phase) return <div>Phase not found</div>;
 
   return (
     <StyledMain>
       <StyledBackgroundSection>
         <StyledContentSection>
-          <PasePageHeader {...surface} />
+          <PasePageHeader {...phase} />
```
Comment (Noggling) [inline client/packages/portal-pages/src/pages/phase-page/PhasePage.tsx:21]: view?

**Review by Noggling — DISMISSED** [review]

**Review by Noggling — APPROVED** [review]

**Pull request merged**